GitUI/CommandsDialogs/RevisionFileTreeControl.cs

```diff
@@ -351,35 +356,59 @@ private void tvGitTree_AfterSelect(object sender, TreeViewEventArgs e)
             Task ViewItem()
             {
                 return e.Node?.Tag is GitItem gitItem
-                    ? ViewGitItemAsync(gitItem)
-                    : Task.CompletedTask;
+                    ? ShowGitItemAsync(gitItem)
+                    : ClearOutputAsync();
             }
+        }
 
-            Task ViewGitItemAsync(GitItem gitItem)
+        private Task ShowGitItemAsync(GitItem gitItem)
+        {
+            switch (gitItem.ObjectType)
             {
-                switch (gitItem.ObjectType)
-                {
-                    case GitObjectType.Blob:
-                    case GitObjectType.Commit:
+                case GitObjectType.Blob:
                     {
-                        GitItemStatus file = new(name: gitItem.FileName)
+                        if (!blameToolStripMenuItem1.Checked)
                         {
-                            IsTracked = true,
-                            TreeGuid = gitItem.ObjectId,
-                            IsSubmodule = gitItem.ObjectType == GitObjectType.Commit
-                        };
+                            return ViewGitItemAsync(gitItem);
+                        }
 
-                        return FileText.ViewGitItemAsync(file, gitItem.ObjectId);
+                        BlameControl.Visible = true;
+                        FileText.Visible = false;
+                        BlameControl.LoadBlame(_revision, null, gitItem.FileName, _revisionGrid, null, FileText.Encoding);
+                        return Task.CompletedTask;
                     }
 
-                    default:
+                case GitObjectType.Commit:
                     {
-                        return FileText.ViewTextAsync("", "");
+                        return ViewGitItemAsync(gitItem);
                     }
-                }
+
+                default:
+                    return ClearOutputAsync();
+            }
+
+            Task ViewGitItemAsync(GitItem gitItem)
+            {
+                GitItemStatus file = new(name: gitItem.FileName)
```

Comment:
Just out of curiosity, what is this syntax, this is C# 9.0 feature?

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9#fit-and-finish-features
Compact code - clearer and more compact than `var`